plugins/vpp/srplugin/vppcalls/vpp1901/srv6.go

```diff
@@ -0,0 +1,598 @@
+func (h *SRv6VppHandler) DeleteLocalSid(sidAddr net.IP) error {
+	return h.addDelLocalSid(true, sidAddr, nil)
+}
+
+func (h *SRv6VppHandler) addDelLocalSid(deletion bool, sidAddr net.IP, localSID *srv6.LocalSID) error {
+	h.log.WithFields(logging.Fields{"localSID": sidAddr, "delete": deletion, "FIB table ID": h.fibTableID(localSID), "end function": h.endFunction(localSID)}).
+		Debug("Adding/deleting Local SID", sidAddr)
+	if !deletion && localSID.GetEndFunction_AD() != nil {
+		return h.addSRProxy(sidAddr, localSID)
+	}
+	req := &sr.SrLocalsidAddDel{
+		IsDel:    boolToUint(deletion),
+		Localsid: sr.Srv6Sid{Addr: []byte(sidAddr)},
+	}
+	if !deletion {
+		req.FibTable = localSID.FibTableId // where to install localsid entry //TODO meaning change? see proto model TODO
+		if err := h.writeEndFunction(req, sidAddr, localSID); err != nil {
+			return err
+		}
+	}
+	reply := &sr.SrLocalsidAddDelReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+	if reply.Retval != 0 {
+		return fmt.Errorf("vpp call %q returned: %d", reply.GetMessageName(), reply.Retval)
+	}
+
+	h.log.WithFields(logging.Fields{"localSID": sidAddr, "delete": deletion, "FIB table ID": h.fibTableID(localSID), "end function": h.endFunction(localSID)}).
+		Debug("Added/deleted Local SID ", sidAddr)
+
+	return nil
+}
+
+// addSRProxy adds local sid with SR-proxy end function (End.AD). This functionality has no binary API in VPP, therefore
+// CLI commands are used (VPE binary API that calls VPP's CLI).
+func (h *SRv6VppHandler) addSRProxy(sidAddr net.IP, localSID *srv6.LocalSID) error {
+	// get VPP-internal names of IN and OUT interfaces
+	names, err := h.interfaceNameMapping()
+	if err != nil {
+		return fmt.Errorf("can't convert interface names from etcd to VPP-internal interface names:%v", err)
+	}
+	outInterface, found := names[localSID.GetEndFunction_AD().OutgoingInterface]
+	if !found {
+		return fmt.Errorf("can't find VPP-internal name for interface %v (name in etcd)", localSID.GetEndFunction_AD().OutgoingInterface)
+	}
+	inInterface, found := names[localSID.GetEndFunction_AD().IncomingInterface]
+	if !found {
+		return fmt.Errorf("can't find VPP-internal name for interface %v (name in etcd)", localSID.GetEndFunction_AD().IncomingInterface)
+	}
+
+	// add SR-proxy using VPP CLI
+	data, err := h.RunCli(fmt.Sprintf("sr localsid address %v behavior end.ad nh %v oif %v iif %v", sidAddr, localSID.GetEndFunction_AD().ServiceAddress, outInterface, inInterface))
+	if err != nil {
+		return err
+	}
+	if len(strings.TrimSpace(string(data))) > 0 {
+		return fmt.Errorf("addition of dynamic segment routing proxy failed by returning nonblank space text in CLI: %v", string(data))
+	}
+	return nil
+}
+
+// interfaceNameMapping dumps from VPP internal names of interfaces and uses them to produce mapping from ligato interface names to vpp internal names.
+func (h *SRv6VppHandler) interfaceNameMapping() (map[string]string, error) {
```

Comment:
The mapping between interface names and their IDs is already exposed by the ifplugin: https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/ifplugin/ifplugin_api.go#L27
Example where it is being used: https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/l3plugin/vppcalls/arp_vppcalls.go#L27-L30